File: core/causal_intelligence/runtime_semantic_equilibrium_engine.py
from __future__ import annotations

from typing import Any, Dict


def assess_semantic_equilibrium(
    runtime_ir: Dict[str, Any],
) -> Dict[str, Any]:
    equilibrium = runtime_ir.get("runtime_equilibrium", {})
    if not isinstance(equilibrium, dict):
        entropy = runtime_ir.get("runtime_entropy", {})
        score = int(entropy.get("entropy_score", 0))
        state = "stable" if score < 100 else "unstable"
    else:
        state = equilibrium.get("equilibrium", "unknown")
    pressure = runtime_ir.get("execution_pressure", {})
    pressure_score = int(pressure.get("pressure_score", 0))
    balanced = state == "stable" and pressure_score < 5000
    return {
        "semantic_equilibrium": state,
        "balanced": balanced,
    }

File: src/causal_intelligence/runtimeSemanticEquilibriumEngine.ts
/**
 * Converted from Python: core/causal_intelligence/runtime_semantic_equilibrium_engine.py
 * @generated — WebWeaveX python→javascript library port
 */

import * as py from "../runtime/pyCompat.js";

export function assessSemanticEquilibrium(runtime_ir: any): any {
  var equilibrium: any = py.get(runtime_ir, "runtime_equilibrium", {});
  if (!((equilibrium !== null && typeof equilibrium === "object" && !Array.isArray(equilibrium) && !(equilibrium instanceof Set) && !(equilibrium instanceof Map)))) {
    var entropy: any = py.get(runtime_ir, "runtime_entropy", {});
    var score: any = py.toInt(py.get(entropy, "entropy_score", 0));
    var state: any = ((score < 100) ? "stable" : "unstable");
  } else {
    state = py.get(equilibrium, "equilibrium", "unknown");
  }
  var pressure: any = py.get(runtime_ir, "execution_pressure", {});
  var pressure_score: any = py.toInt(py.get(pressure, "pressure_score", 0));
  var balanced: any = py.and2(py.eq(state, "stable"), () => ((pressure_score < 5000)));
  return {"semantic_equilibrium": state, "balanced": balanced};
}
